src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -141,6 +141,8 @@ protected virtual void Dispose(bool disposing)
         /// <param name="TelemetryEvent"> Telemetry event. </param>
         public static void EmitTelemetryEvent(TelemetryEvent TelemetryEvent)
         {
+            Debug.Assert(NuGetTelemetryService != null, $"{nameof(NuGetTelemetryService)} should not be null!");
```

Comment:
This is going to break command line tool & tests in debug configuration.

Reply:
I put this here to detect when we're trying to log telemetry but the service is `null` which is a bug.  Is there a case when the service would be null when running tests?

Reply:
yes, every test that doesn't validate telemetry could have this static as null.  Also, it's by design that this is null for CLI tools like nuget.exe and msbuild restore, but our shared restore code will still call into this method. Maybe it's a design bug, but this is going to cause problems for anyone trying to debug nuget.exe or msbuild restore.

Reply:
I agree we should remove this.

Reply:
So far I don't think its been hit in unit tests, but I'll remove it.  I mostly put it in there to catch any scenario where it wasn't set when I was debugging but I replaced it with a conditional breakpoint instead